pkg/controllers/coscheduling/podgroup_controller.go

```diff
@@ -47,16 +66,153 @@ type PodGroupReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *PodGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := log.FromContext(ctx)
+	pg := &schedv1alpha1.PodGroup{}
+	err := r.Get(ctx, req.NamespacedName, pg)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			log.V(5).Info("Pod group has been deleted")
+			return ctrl.Result{}, nil
+		}
+		log.V(3).Error(err, "Unable to retrieve pod group")
+		return ctrl.Result{}, err
+	}
+
+	if pg.Status.Phase == schedv1alpha1.PodGroupFinished ||
+		pg.Status.Phase == schedv1alpha1.PodGroupFailed {
+		return ctrl.Result{}, nil
+	}
+	// If startScheduleTime - createTime > 2days,
+	// do not enqueue again because pod may have been GCed
+	if pg.Status.Scheduled == pg.Spec.MinMember && pg.Status.Running == 0 &&
+		pg.Status.ScheduleStartTime.Sub(pg.CreationTimestamp.Time) > 48*time.Hour {
+		return ctrl.Result{}, nil
+	}
+
+	pgCopy := pg.DeepCopy()
+	podList := &corev1.PodList{}
+	err = r.List(ctx, podList,
+		client.MatchingLabelsSelector{
+			Selector: labels.Set(map[string]string{
+				schedv1alpha1.PodGroupLabel: pgCopy.Name}).AsSelector(),
+		})
+	if err != nil {
```

Comment:
```suggestion
        labelSelector := client.MatchingLabelsSelector{
			Selector: labels.Set(map[string]string{
				schedv1alpha1.PodGroupLabel: pg.Name}).AsSelector(),
		}
	if err = r.List(ctx, podList, labelSelector); err != nil {
```

Reply:
the `labelSelector` will only be used once in this `List`, why create a tmp var for it?

Reply:
Yeah, you're right it's better to make func for that to make this more concise :
```go
func genLabelSelectorFor(name string) client.MatchingLabelsSelector{
    return client.MatchingLabelsSelector{
			Selector: labels.Set(map[string]string{
				schedv1alpha1.PodGroupLabel: name}).AsSelector(),
    }
}
...
if err = r.List(ctx, podList, genLabelSelectorFor(pg.Name)); err != nil {
```